ambry-store/src/main/java/com.github.ambry.store/StorageManager.java

```diff
@@ -428,5 +429,35 @@ public void onPartitionBecomeLeaderFromStandby(String partitionName) {
     public void onPartitionBecomeStandbyFromLeader(String partitionName) {
       // no op
     }
+
+    @Override
+    public void onPartitionBecomeInactiveFromStandby(String partitionName) {
+      // check if partition exists on current node
+      ReplicaId replica = partitionNameToReplicaId.get(partitionName);
+      // if replica is null that means partition is not on current node (this shouldn't happen unless we use server admin
+      // tool to remove the store before initiating decommission on this partition). We throw exception in this case.
+      if (replica != null) {
+        Store localStore = getStore(replica.getPartitionId());
+        if (localStore != null) {
+          // 1. set state to INACTIVE
+          localStore.setCurrentState(ReplicaState.INACTIVE);
+          logger.info("Store {} is set to INACTIVE", partitionName);
+          // 2. disable compaction on this store
+          if (!controlCompactionForBlobStore(replica.getPartitionId(), false)) {
+            logger.error("Failed to disable compaction on store {}", partitionName);
+            throw new StateTransitionException("Couldn't disable compaction on replica " + replica.getReplicaPath(),
+                StateTransitionException.TransitionErrorCode.ReplicaOperationFailure);
```

Comment:
Minor: consider renaming ReplicaOperationFailure or using one of the other error codes.

Reply:
Right, I will probably change it to `ReplicaNotFound` because it is the only reason disabling compaction may fail.